snapshots/devmapper/snapshotter_test.go

```diff
@@ -165,3 +140,85 @@ func TestMkfsXfsNonDefault(t *testing.T) {
 	err := mkfs(ctx, "xfs", "noquota", "")
 	assert.ErrorContains(t, err, `mkfs.xfs couldn't initialize ""`)
 }
+
+func TestMultipleXfsMounts(t *testing.T) {
+	testutil.RequiresRoot(t)
+
+	logrus.SetLevel(logrus.DebugLevel)
+
+	ctx := context.Background()
+	ctx = namespaces.WithNamespace(ctx, "testsuite")
+	tempDir, err := os.MkdirTemp("", "snapshot-suite-usage")
+	assert.NilError(t, err)
+	defer os.RemoveAll(tempDir)
+
+	poolName := fmt.Sprintf("containerd-snapshotter-suite-pool-%d", time.Now().Nanosecond())
+	config := &Config{
+		RootPath:       tempDir,
+		PoolName:       poolName,
+		BaseImageSize:  "16Mb",
+		FileSystemType: "xfs",
+	}
+	snapshotter, closer, err := createSnapshotter(ctx, t, config)
+	assert.NilError(t, err)
+	defer closer()
+
+	var (
+		sizeBytes   int64 = 1048576 // 1MB
+		baseApplier       = fstest.Apply(fstest.CreateRandomFile("/a", 12345679, sizeBytes, 0777))
+	)
```

Comment:
We don't need them.

Reply:
Why? `baseApplier` is needed to create a snapshot.

Reply:
I think (not 100% sure, sorry) that Prepare() returns a XFS-formatted devmapper device. So mounting two of them is enough to reproduce the original issue. Would it work?

Reply:
Nope. we have to prepare an additional snapshot from a base one.